### PR TITLE
Added documentation for deploy groups

### PR DESCRIPTION
**Pull request by mjksmith**
You can see the deploy groups rst here: http://rst.ninjs.org/?n=3482021a3416f45d4b1aee08e9713b4e&theme=nature

### CODE REVIEW

File: docs/source/deploy_groups.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+New service
+-----------
+
+When you run ``paasta fsm`` to generate the boilerplate configs for your paasta service, the default configs will have four deploy groups: ``dev.everything``, ``stage.everything``, ``prod.canary`` and ``prod.non_canary``. You can change these deploy groups around to suit your needs, and also change their deploy order by editing ``deploy.yaml``. Then, edit your soa configs and generate your pipline like normal.
+
+Existing service
+----------------
+
+Edit your ``deploy.yaml`` and your ``marathon-*.yaml`` and ``chronos-*.yaml`` files to use deploy groups. For each instance in a deploy group, specify a ``deploy_group`` parameter in their config file. Then, remove their deployment steps from ``deploy.yaml`` and replace it with a single deploy step for the deploy group. Finally, update your jenkins workflow by deleting and re-creating it like normal.
```
Comment (solarkennedy) [inline docs/source/deploy_groups.rst]: Are you sure this order is correct? Seems like the proper order is:
* Add deploy steps in deploy.yaml
* Run the pipeline to mark them
* Edit marathon/chronos yamls and remove old steps from deploy.yaml

File: docs/source/deploy_groups.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+What if I don’t want to use deploy groups on my existing service?
+-----------------------------------------------------------------
+
+No changes are required -- your service should work as-is.
```
Comment (solarkennedy) [inline docs/source/deploy_groups.rst]: Use this opportunity to reiterate that the default deploy_group is `cluster.instance`

File: docs/source/yelpsoa_configs.rst [hunk 3]
```diff
@@ -134,6 +134,11 @@ instance MAY have:
    logic will assume that there are no instances in the blacklisted region,
    implying that we should expect all 10 in the non-blacklisted region.
 
+ * ``deploy_group``: A string identifying what deploy group this instance belongs
+   to. The ``instancename`` parameter in ``deploy.yaml`` refererences this value
+   to determine the order in which to build & deploy deploy groups. Defaults to
+   ``clustername.instancename``.
```
Comment (solarkennedy) [inline docs/source/yelpsoa_configs.rst]: point to your other doc here for further reading

Comment (solarkennedy) [inline docs/source/yelpsoa_configs.rst]: Chronos jobs use this too, so be sure to make that explicit and point to this

File: docs/source/yelpsoa_configs.rst [hunk 4]
```diff
@@ -279,6 +284,11 @@ Each job configuration MAY specify the following options:
     <https://mesos.github.io/chronos/docs/api.html#constraints>`_ for more
     information.
 
+  * ``deploy_group``: A string identifying what deploy group this instance belongs
```
Comment (solarkennedy) [inline docs/source/yelpsoa_configs.rst]: don't copy paste, just reference the above

File: docs/source/yelpsoa_configs.rst [hunk 5]
```diff
@@ -138,6 +138,11 @@ instance MAY have:
    logic will assume that there are no instances in the blacklisted region,
    implying that we should expect all 10 in the non-blacklisted region.
 
+  * ``deploy_group``: A string identifying what deploy group this instance belongs
+    to. The ``instancename`` parameter in ``deploy.yaml`` refererences this value
+    to determine the order in which to build & deploy deploy groups. Defaults to
+    ``clustername.instancename``. See the deploy group doc for more information.
```
Comment (solarkennedy) [inline docs/source/yelpsoa_configs.rst]: do an actual html link for 'deploy group'

File: docs/source/deploy_groups.rst [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+Existing service
+----------------
+
+* Plan out how you want to divide your service into deploy groups
+
+* Append deployment steps for these deploy groups to the end of your service's ``deploy.yaml`` file
+  
+* Push your changes to the soa configs repo
+  
+* Regenerate your jenkins workflow by running the ``paasta generate-pipeline`` command
```
Comment (mjksmith) [inline docs/source/deploy_groups.rst]: Right now these docs are pretty yelp-specific in that they mention paasta generate-pipeline as the default way to migrate to deploy groups.

Should we put this elsewhere (somewhere yelp-specfic) or just make these instructions more general? (or both or neither)

Comment (solarkennedy) [inline docs/source/deploy_groups.rst]: na, just leave them

**Comment by solarkennedy** [comment]
ship

**Pull request merged**